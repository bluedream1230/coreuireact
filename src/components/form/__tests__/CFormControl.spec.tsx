import * as React from 'react'
import { render /* ,screen */, fireEvent /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CFormControl } from '../../../index'

test('loads and displays CFormControl component', async () => {
  const { container } = render(<CFormControl />)
  expect(container).toMatchSnapshot()
})

test('CFormControl customize', async () => {
  const { container } = render(
    <CFormControl
      className="bazinga"
      component="h3"
      disabled={true}
      plainText={true}
      readOnly={true}
      size="lg"
      type="color"
      value="value"
    />,
  )
  expect(container).toMatchSnapshot()
  expect(container.firstChild).toHaveClass('bazinga')
  expect(container.firstChild).toHaveClass('form-control-plaintext')
  expect(container.firstChild).toHaveClass('form-control-color')
  expect(container.firstChild).toHaveClass('form-control-lg')
})

test('CFormControl customize classNameParent', async () => {
  const { container } = render(
    <CFormControl className="bazinga" classNameParent="classNameParent" />,
  )
  expect(container).toMatchSnapshot()
})

test('CFormControl change input', async () => {
  jest.useFakeTimers()
  const onChange = jest.fn()
  render(<CFormControl onChange={onChange} />)
  expect(onChange).toHaveBeenCalledTimes(0)
  const input = document.querySelector('input')
  if (input !== null) {
    fireEvent.change(input, { target: { value: 'bazinga' } })
  }
  expect(onChange).toHaveBeenCalledTimes(1)
  if (input !== null) {
    fireEvent.change(input, { target: { value: '2' } })
  }
  expect(onChange).toHaveBeenCalledTimes(2)
})
