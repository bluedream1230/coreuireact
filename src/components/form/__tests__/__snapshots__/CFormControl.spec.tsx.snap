// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CFormControl customize 1`] = `
<div>
  <h3
    class="form-control-plaintext form-control-color form-control-lg bazinga"
    disabled=""
    readonly=""
    value="value"
  />
</div>
`;

exports[`CFormControl customize classNameParent 1`] = `
<div>
  <input
    class="classNameParent"
    type="text"
  />
</div>
`;

exports[`loads and displays CFormControl component 1`] = `
<div>
  <input
    class="form-control"
    type="text"
  />
</div>
`;
